### PR TITLE
Support webworkers

### PR DESCRIPTION
**Pull request by schehata**

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -2,9 +2,15 @@ import GenericConfig from './platform/generic';
 import VercelConfig from './platform/vercel';
 import NetlifyConfig from './platform/netlify';
 
+declare global {
+  var Cloudflare: string;
```
Comment (bahlo) [inline src/config.ts:6]:
```suggestion
  var WebSocketPair: any; // Non-standard global only used on Cloudflare: https://developers.cloudflare.com/workers/runtime-apis/websockets
```

Sorry, I don't think there's a global `Cloudflare` object—best guess right now is checking for this non-standard global.

Comment (DaniFoldi) [reply src/config.ts]: Hi :wave:

Cloudflare Workers, and Pages Functions define `navigator.userAgent` as `'Cloudflare-Workers'`: https://developers.cloudflare.com/workers/runtime-apis/web-standards/#navigatoruseragent

This is probably the most future-proof way to detect running in Workers/Pages.

Comment (bahlo) [reply src/config.ts]: Oh, that's cool—thanks @DaniFoldi 🙏

**Comment by schehata** [comment]
@arne `WebSocketsPair` doesn't work. `WorkerGlobalScope` works fine.

File: src/platform/generic.ts [hunk 2]
```diff
@@ -24,11 +24,11 @@ export default class GenericConfig implements Provider {
   }
 
   getLogsEndpoint(): string {
-    return this.isBrowser ? `${this.proxyPath}/logs` : this.getIngestURL(EndpointType.logs);
+    return isBrowser ? `${this.proxyPath}/logs` : this.getIngestURL(EndpointType.logs);
   }
 
   getWebVitalsEndpoint(): string {
-    return this.isBrowser ? `${this.proxyPath}/logs` : this.getIngestURL(EndpointType.webVitals);
+    return isBrowser ? `${this.proxyPath}/logs` : this.getIngestURL(EndpointType.webVitals);
```
[the remark this answers is not in the document]
Comment (schehata) [reply src/platform/generic.ts:31]: hmmm, but we don't have a web-vitals endpoint anyway.

[the remark this answers is not in the document]
Comment (bahlo) [reply src/platform/generic.ts:31]: no but we have `/_axiom/web-vitals` proxied

**Review by bahlo — APPROVED** [review]

**Pull request merged**